api/src/beacons/beacons.service.ts

```diff
@@ -37,20 +37,38 @@ export class BeaconsService {
     latitude,
     longitude,
     maxDistance,
+    tags
   }: FindBeaconsDto): Promise<Beacon[]> {
-    return this.beaconModel
-      .find({
-        location: {
-          $near: {
-            $geometry: {
-              type: 'Point',
-              coordinates: [latitude, longitude],
+    if (tags && tags.length > 0) {
+      return this.beaconModel.find({$and: [
+        {tags: {$in: tags}},
+        {location:
+          {
+            $near: {
+              $geometry: {
+                type: 'Point',
+                coordinates: [latitude, longitude],
+              },
+              $maxDistance: maxDistance,
+            },
+          }
+        }]
+      }).exec();
+    } else {
+      return this.beaconModel
+        .find({
+          location: {
+            $near: {
+              $geometry: {
+                type: 'Point',
+                coordinates: [latitude, longitude],
+              },
+              $maxDistance: maxDistance,
             },
-            $maxDistance: maxDistance,
           },
-        },
-      })
-      .exec();
+        })
+        .exec();
```

Comment:
I think you can replace this with this.
`return this.beaconModel
  .find({
    location: {
      $near: {
        $geometry: {
          type: 'Point',
          coordinates: [latitude, longitude],
        },
        $maxDistance: maxDistance,
      },
    },
    // Add a condition to filter based on tags if tags parameter is provided
    ...(tags && tags.length > 0 ? { tag: { $in: tags } } : {}), 
  })
  .exec();`